saiplayer/SaiPlayer.cpp

```diff
@@ -2522,17 +2629,17 @@ int SaiPlayer::replay()
 
     SWSS_LOG_NOTICE("using file: %s", filename.c_str());
 
-    std::ifstream infile(filename);
```

Comment:
why this logic is changed ? not need for this variable to be member of class

Reply:
to use m_infile in processBulk method

Reply:
i see that, but why the reason ? since this is local variable no need to upgrade it to member

Reply:
SaiPlayer operates on input stream, the input stream is feeding multiple functions with data throuhgout this operation. From that point of view the lifetime of the input stream needs to match the lifetime of SaiPlayer, thus fits to be a member variable. I did this change for ergonomics primaraly - not having to pass the input stream explicitelly everywhere it is needed in downstream code. Besides, passing an input stream explicitelly implies a member function may operate on different input streams throughout SaiPlayer's lifetime which is not the case.
I can make a function parameter though if you prefer so.